File: textScript.js


let fs = require("fs")
const path = require("path")
var i = 0

function searchJies(lines){
    let jies = []
    while(1){
        let text = lines[i]
        if(text.includes("第") && text.includes("节")){
            text = text.replace(/\s*/g,"")
            i += 2
            let fangjies = searchFangjies(lines)
            let jie = {
                jie:text,
                zucheng:fangjies
            }
            jies.push(jie)
        }else{
            return jies
        }
    }
}

function searchFangjies(lines){
    let fangjies = []
    while(1){
        let text = lines[i]
        if(i == 640){
            return fangjies
        }
        if(text.includes("组成")){
            text = text.replace(/\s*/g,"")
            i += 2
            fangjies.push(text)
        }else{
            return fangjies
        }
    }
}


fs.readFile('zucheng.txt', 'utf-8', (err, data)=>{
    // console.log(data);
    let lines = data.split(/\r?\n/)
    let flag = 1
    let mydata = []
    while(flag == 1){
        while(1){
            let text = lines[i]
            if(text.includes("章") && text.includes("第")){
                text = text.replace(/\s*/g,"")
                i += 2
                jies = searchJies(lines)
                let zhang = {
                    zhang:text,
                    jies:jies
                }
                mydata.push(zhang)
            }else{
                let dataJson = JSON.stringify(mydata)
                console.log(dataJson);
                let file = path.resolve(__dirname,'./liziJson.txt')
                fs.writeFile(file, dataJson,{encoding:'utf-8'}, err=>{
                    console.log(err);
                })
                fs.close()
                break
            }
        }
    }

})

File: textScript.py

import json
import linecache
from math import fabs

file = "lizi.txt"

data = {}
def searchJies(i):
    i = i
    jies = []
    while(1):
        
        text = linecache.getline(file, i)
        
        if "第" in text and "节" in text:
            i = i + 2

            i, fangjies = searchFangJies(i)

            text = text.replace("\n", "")
            jie = {
                "jie":text,
                "lizi":fangjies
            }
            jies.append(jie)
        else:
            return i, jies

    

def searchFangJies(i):
    i = i
    fangjies = []
    while(1):
        
        text = linecache.getline(file, i)

        if "例" in text :
            i = i + 2

            text = text.replace("\n", "")
            fangjies.append(text)
        else:
            return i, fangjies
flag = 1
while flag == 1:
    data = []
    i = 1
    while(1):
        
        text = linecache.getline(file, i)
        if "章" in text and "第" in text:
            i = i + 2
            i, jies = searchJies(i)

            text = text.replace("\n", "")
            zhang = {
                "zhang": text,
                "jies":jies
            }
            data.append(zhang)
        else:
            # print(json.dumps(data,ensure_ascii=False))

            print(data)
            print(i)
            flag = 0
            break
